src/server/serverUtils/statusBarNotifier.ts

```diff
@@ -49,13 +84,25 @@ export class StatusBarNotifier extends Disposable {
 	}
 
 	/**
-	 * @description called to notify that the server shut down.
+	 * @description called to notify that all of the servers are off
 	 */
 	public ServerOff(): void {
 		this._on = false;
 		this._statusBar.hide();
 	}
 
+	/**
+	 * @description called to notify that a server shut down.
+	 */
+	public RemoveServer(uri: vscode.Uri | undefined): void {

```

Comment:
I would structure this so that the StatusBarNotifier can listen to some event and update itself accordingly, instead of having to call into it from other code. Think of it this way, you can have a model with some event fired on update, and other features can be built on top of it. That reduces the amount of code that has to know about a StatusBarNotifier